opal/runtime/opal_init.c

```diff
@@ -520,7 +521,7 @@ opal_init_util(int* pargc, char*** pargv)
  * versions of memcpy correctly configured.
  */
 static mca_base_framework_t *opal_init_frameworks[] = {
-    &opal_hwloc_base_framework, &opal_memcpy_base_framework, &opal_memchecker_base_framework,
+    &opal_hwloc_base_framework, &opal_memcpy_base_framework, &opal_threads_base_framework, &opal_memchecker_base_framework,
```

Comment:
Are no mutexes/threads/condition variables used before this framework is initialized?

Reply:
@jsquyres synchronization variables can be declared statically as well as initialized by opal. Static initialization and inlining synchronization functions are two performance optimizations that break the mca abstraction.

Reply:
Understood.  My real question was: you inserted this framework as 3rd in the sequence.  Do we know that that is sufficient?  Or does it need to be first?  And is first even high enough -- or are there locks/etc. used before any frameworks are initialized?

Reply:
@jsquyres I don't, but I believe these synchronization MCAs existed before this pull request and the priority was the same then. This is beyond my current understanding of OpenMPI though. Who knows the most about this?

Reply:
I don't know if anyone knows this -- it's probably going to have to be determined experimentally...

Reply:
It looks like the threads framework is still 3rd in the list.  Did you dig into this and find out that it still needs to be 3rd?